### DISCUSSION TITLE
DuckDB seems slow to represent the results as a string

### DISCUSSION THREAD

**Discussion by datapythonista** [duckdb/duckdb#10705] [General]
I'm comparing different Python libraries for a particular task of processing a CSV file. The results with DuckDB seem a bit strange, as the result of `duckdb.sql()` takes around 100ms which is consistent with Polars, but if I convert the result to a string to visualize it the result becomes 800ms (the result is a single scalar). I don't know how this works internally, but feels like `.sql()` is already executing the query, as 100ms seems too long to just create the execution plan. If that's the case going from 100ms to 800ms just to convert DuckDB's internal representation to a string for a single scalar seems a lot.

I don't see a `.collect()` method or anything similar, other than methods to convert to other formats, I'm not sure if I should be using something else instead, but doesn't seem like it.

You can check the code of what I'm doing here: https://github.com/datapythonista/bench_csv/blob/master/python_benchmarks/30_duckdb_sql.py

And the results of the benchmarks to see the comparison with other libraries here: https://github.com/datapythonista/bench_csv/tree/master

**Comment by cmdlineluser** [comment]
> I don't see a .collect() method or anything similar

They are listed here: https://duckdb.org/docs/api/python/overview.html#result-conversion

As for `duckdb.sql()` - it doesn't execute the query in full:

> The query is **not executed** until the result is fetched or requested to be printed to the screen.

ref: https://duckdb.org/docs/api/python/overview.html#basic-api-usage

**Comment by Tishj** [comment]
I dont think this benchmark makes sense, we're not an OLTP database, this is not a usecase we optimize for

**Comment by datapythonista** [comment]
Thanks @cmdlineluser, this is very helpful!

@Tishj would you mind expanding on why the query is not an OLAP use case? It counts the number of elements in the dataset that are closer to a point in an 8-dumensional space. Seems like a clear OLAP use case to me, I must be missing something.

**Comment by Tishj** [comment]
> If that's the case going from 100ms to 800ms just to convert DuckDB's internal representation to a string for a single scalar seems a lot.

I thought you were measuring conversion time, apologies
